### PR TITLE
[cd] Deploy concourse using instance refresh

### PR DESCRIPTION
**Pull request by philandstuff**
Currently, the deployment pipeline manages to use concourse to deploy
concourse by scaling itself out and in again.  This could be made
simpler by using the new [Instance Refresh feature][1].  This means
that we don't have to change the desired number of workers, or re-run
terraform, or anything.

[1]: https://aws.amazon.com/blogs/compute/introducing-instance-refresh-for-ec2-auto-scaling/

https://trello.com/c/Fkc86srz

### CODE REVIEW

**Comment by philandstuff** [comment]
also I'm not sure if the bash conditionals work, I don't know if `[[` supports `-a` like this

File: reliability-engineering/pipelines/tasks/asg-wait-for-refresh.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  args:
+  - -euo
+  - pipefail
+  - -c
+  - |
+    function refresh_fetch_info() {
+      aws autoscaling describe-instance-refreshes \
+        --auto-scaling-group-name "${ASG_NAME}" \
+        --instance-refresh-ids "$(cat refresh/id)" \
+        | jq "InstanceRefreshes[]" \
```
Comment (Krenair) [inline reliability-engineering/pipelines/tasks/asg-wait-for-refresh.yml:22]:
```suggestion
        | jq ".InstanceRefreshes[]" \
```
?

File: reliability-engineering/pipelines/concourse-deployer.yml [hunk 2]
```diff
@@ -169,91 +168,72 @@ jobs:
...
       params:
-        DEPLOYMENT_NAME: ((deployment_name))
-    on_failure:
-      put: infra # try to scale back in on failure to stay in sync
-      attempts: 10
+        ASG_NAME: ((deployment_name))-concourse-web
+    - task: refresh-main-team-workers-asg
+      file: tech-ops/reliability-engineering/pipelines/tasks/asg-refresh-instances.yml
+      output_mapping:
+        refresh: main-team-worker-refresh
```
Comment (philandstuff) [inline reliability-engineering/pipelines/concourse-deployer.yml:187]: we write to the `refresh` volume and try to read from it in a separate job.  To fix this, combine the `refresh-instances` and `wait-for-refresh` jobs into one job with different steps.

**Comment by philandstuff** [comment]
@Krenair updated with feedback, also did some extra tidying commits

**Comment by risicle** [comment]
I'd be in favour of this just from the amount of bash it removes.

**Review by Krenair — APPROVED** [review]

**Pull request merged**